### PR TITLE
[ir_builder] unroll const loops with extent=1

### PR DESCRIPTION
**Pull request by xqdan**
Thanks for contributing to TVM!   Please refer to guideline https://docs.tvm.ai/contribute/ for useful information and tips. After the pull request is submitted, please request code reviews from others in the community.

### CODE REVIEW

**Comment by tqchen** [comment]
Please request reviews from https://github.com/dmlc/tvm/blob/master/CONTRIBUTORS.md#reviewers

**Comment by xqdan** [comment]
@yzhliu @were @kevinthesun please review, thanks!

**Comment by tqchen** [comment]
I am not sure if we really want this eager behavior in the ir builder, as it complicates the implementation and makes the user's intention a bit ambiguous.

File: python/tvm/ir_builder.py [hunk 1]
```diff
@@ -197,6 +197,16 @@ def for_range(self, begin, end, name="i", dtype="int32", for_type="serial"):
         self._seq_stack.append([])
         loop_var = _api.var(name, dtype=dtype)
         extent = end if begin == 0 else _pass.Simplify(end - begin)
+
+        if for_type == "serial":
```
Comment (kevinthesun) [inline python/tvm/ir_builder.py]: I think this implicit modification makes thing a bit opaque to developer. Also will it always be better to do such a transformation?

**Comment by xqdan** [comment]
the reason we do this is that, we can eliminate the backward dependence for false loops(with extent=1) when we inject sync in coproc_sync pass. Actually we improve the coproc_sync to support injecting syncs across the pipelines compared to inject just for neighbors of pipelines, and also we inject syncs based on dependence compared to only consider if they are from different pipelines. that's why we got backward dependence for loops. so these loops with extent = 1 hurt performance badly.

I understand the reason you mentioned. another solution i come up with is that add a new pass to mark all these false loop as unrolled, and also expose a flag to users.

Anyway, I can close these and send a new one based on our discussion result.

**Comment by tqchen** [comment]
I think we should close this. We can, however, enforce unroll of loop with extent=1 in the loop unrolling pass

**Pull request closed without merging**